File: code/backend/php/src/Core/Domain/Matches/MatchDetail.php
<?php

namespace Games\Paths\Core\Domain\Matches;

class MatchDetail
{
    /**
     * @param MatchLocationState[] $locations
     * @param MatchRegistryEntry[] $registry
     * @param MatchEventOption[] $events
     * @param MatchEventOption[] $choices
     * @param CharacterInstanceInfo[] $players
     */
    public function __construct(
        public readonly MatchSummary $match,
        public readonly ?int $currentLocationId = null,
        public readonly ?string $currentLocationUuid = null,
        public readonly ?string $currentLocationName = null,
        public readonly array $locations = [],
        public readonly array $registry = [],
        public readonly array $events = [],
        public readonly array $choices = [],
        public readonly array $players = []
    ) {
    }

    public function toArray(): array
    {
        return [
            'match' => $this->match->toArray(),
            'currentLocationId' => $this->currentLocationId,
            'currentLocationUuid' => $this->currentLocationUuid,
            'currentLocationName' => $this->currentLocationName,
            'locations' => array_map(fn($l) => $l->toArray(), $this->locations),
            'registry' => array_map(fn($r) => $r->toArray(), $this->registry),
            'events' => array_map(fn($e) => $e->toArray(), $this->events),
            'choices' => array_map(fn($c) => $c->toArray(), $this->choices),
            'players' => array_map(fn($p) => $p->toSummaryArray(), $this->players),
        ];
    }
}

File: code/backend/java/core/src/main/java/games/paths/core/model/match/MatchDetail.java
package games.paths.core.model.match;

import java.util.ArrayList;
import java.util.List;

/**
 * MatchDetail - Domain model returned by {@code MatchQueryPort.getMatchInfo}.
 *
 * <p>Step 19 — embeds the match summary, the runtime location & registry state,
 * and the events/choices currently available for the player.</p>
 */
public class MatchDetail {

    private MatchSummary match;
    private Long currentLocationId;
    private String currentLocationUuid;
    private String currentLocationName;
    private List<MatchLocationState> locations = new ArrayList<>();
    private List<MatchRegistryEntry> registry = new ArrayList<>();
    private List<MatchEventOption> events = new ArrayList<>();
    private List<MatchEventOption> choices = new ArrayList<>();

    public MatchDetail() {
    }

    public MatchSummary getMatch() { return match; }
    public void setMatch(MatchSummary match) { this.match = match; }

    public Long getCurrentLocationId() { return currentLocationId; }
    public void setCurrentLocationId(Long currentLocationId) { this.currentLocationId = currentLocationId; }

    public String getCurrentLocationUuid() { return currentLocationUuid; }
    public void setCurrentLocationUuid(String currentLocationUuid) { this.currentLocationUuid = currentLocationUuid; }

    public String getCurrentLocationName() { return currentLocationName; }
    public void setCurrentLocationName(String currentLocationName) { this.currentLocationName = currentLocationName; }

    public List<MatchLocationState> getLocations() { return locations; }
    public void setLocations(List<MatchLocationState> locations) {
        this.locations = locations != null ? locations : new ArrayList<>();
    }

    public List<MatchRegistryEntry> getRegistry() { return registry; }
    public void setRegistry(List<MatchRegistryEntry> registry) {
        this.registry = registry != null ? registry : new ArrayList<>();
    }

    public List<MatchEventOption> getEvents() { return events; }
    public void setEvents(List<MatchEventOption> events) {
        this.events = events != null ? events : new ArrayList<>();
    }

    public List<MatchEventOption> getChoices() { return choices; }
    public void setChoices(List<MatchEventOption> choices) {
        this.choices = choices != null ? choices : new ArrayList<>();
    }
}
